### DISCUSSION TITLE
Select font inside font collection

### DISCUSSION THREAD

**Discussion by Bytologic** [ImageMagick/ImageMagick#4540] [Help] [answered]
Hi.

I have been using Magick++ to support Japanese text and their fonts.
To get the right font collection I do something like this:

```
Magick::Image image(Magick::Geometry(800, 300), Magick::Color("white"));
image.font("MS-Gothic-&-MS-UI-GOTHIC-&-MS-PGothic");
image.read("label:japanese text here");
```

This results in the text being rendered in the MS Gothic font.
How would I be able to tell Image Magick to render the text in either MS UI Gothic or MS PGothic font?

**Comment by urban-warrior** [comment]
Try `convert -list font` to view which fonts your instance of ImageMagick supports.  If you need a specific font, you can specify by path, e.g. `/usr/share/fonts/unifont/unifont.ttf`.

**Reply by Bytologic** [reply]
Thank you for your response.

Listing the fonts gave me "MS-Gothic-&-MS-UI-GOTHIC-&-MS-PGothic" as an available font.

The file name of this collection is called msgothic.ttc, as can be seen here: https://docs.microsoft.com/en-gb/typography/font-list/ms-gothic.
I can load this specific file in by using:
```
image.font("C:/Windows/Fonts/msgothic.ttc");
```
However, this leads to the same result where the text is rendered in MS Gothic.
Is there any way for me to specify which specific font needs to be used inside the collection?

**Comment by urban-warrior** [comment, marked as answer]
If you specify the path to the font, that is the font ImageMagick utilizes unless an exception is thrown.  Its possible the font contains all three fonts: Gothic, UI Gothic, and PGothic.  If so, ImageMagick only selects the first font in the bundle.  You may need to unbundle the fonts or use some other method to render your text.

**Reply by Danack** [reply]
"If so, ImageMagick only selects the first font in the bundle. You may need to unbundle the fonts or use some other method to render your text."

Do you happen to have any notes on that please?

**Reply by snibgo** [reply]
Converters for creating TTF files from TTC files are available, for example https://everythingfonts.com/ttc-to-ttf

I have no experience with that or any similar tools.
